### DISCUSSION TITLE
Some PoC feedback

### DISCUSSION THREAD

**Discussion by gkgeorgiev** [camptocamp/oapi-poc#22] [General]
I have some feedback, which of course could be always subjective :)
My Feedback is based mostly on the links of the sample pages provided. 

- **Measurement values air temperature 2 m, Instantaneous Value**  - What I didn't like the most is t he mix between representation and data model. I noticed the JSON contains HTML which is bad separation of concerns and violating the MVC pattern :) Also another thing is the way localisation is handled - IMHO, the API should consider the 'Accept-Language' header (which is a standard) and if there's any locale specific data should be converted/formatted dynamically, without generating different items/links per language. The same comment applies JSON/CSV formats - use Accept header to detect what to return. I also noticed that the encoding is not UTF-8 which IMHO is better to use due to the greater character map and in a long run you can use additional characters without any API consumer being impacted.
- **meteoswiss.radar.precip** - I hope you use  the proper HTTP headers since it's not visible from Swagger, to send the zipped payload.

What I'd also recommend is a bit of consistency in the formats ... some assets are in XML and others  - for instance JSON. In general, more of the HTTP standards should be used and less of proprietary payloads. 
Also as a potential user, I would appreciate some assets documentation since the current API model is too generic and metadata could contain anything. 

In any case,  I really appreciate the effort and the initiative!

**Comment by davidoesch** [comment]
Thanks for the feedback- very helpful and you nailed it.
We will take this into doubt for the next step of the project
